api_app/analyzers_manager/observable_analyzers/ip2whois.py

```diff
@@ -0,0 +1,44 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class Ip2whois(classes.ObservableAnalyzer):
+    base_url: str = "https://api.ip2whois.com/v2"
+    _api_key_name: str
+
+    def get_response(self, payload):
+        return requests.get(self.base_url, params=payload)
+
+    def run(self):
+        try:
+            params = {
+                "key": self._api_key_name,
+                "domain": self.observable_name,
+            }
+
+            location_info = self.get_response(params)
+            location_info.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        response = location_info.json()
+        return response
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockUpResponse({}, 200),
```

Comment:
you could put in this json here a sample of a possible output to make the test more real and to have a record of the expected results keys

Reply:
Hi @mlodic , thanks for the comment, we will try to add one example now. Meanwhile, can we get advise from you on how to solve the errors reported by the backend-tests and DeepSource: Python?

Reply:
yep it works, just now the problem is with the linters, you should apply black and flake as stated in the docs

Reply:
Hi @mlodic , we have pushed some fixes for the issues, let's wait and see.

Reply:
Hi @mlodic , I found that the Build & Tests / backend-tests is failing, can you please help us to see if the issue related to our codes?